### PR TITLE
[GR-1013][GR-1041] Jira Ticket Buttons

### PR DESCRIPTION
**Pull request by asofmosq**
https://jira.oicr.on.ca/browse/GR-1013

https://jira.oicr.on.ca/browse/GR-1041

Where the "file a ticket about these runs" on single-lane pages is visible after the update button is clicked.

### CODE REVIEW

**Review by avarsava — COMMENTED** [review]
I tried filing a ticket with all the runs selected and got HTTP 400 Bad Request "Request header is too large", which makes sense since it's delivering every run through the URL. Is there anything we could do about this? Squashing all the runs to 'all_runs' like we do for logging would cause a bug report to lose state when the data updates

**Comment by callunity** [comment]
> I tried filing a ticket with all the runs selected and got HTTP 400 Bad Request "Request header is too large", which makes sense since it's delivering every run through the URL. Is there anything we could do about this? Squashing all the runs to 'all_runs' like we do for logging would cause a bug report to lose state when the data updates

I can't imagine a meaningful reason to file a ticket with all runs, so it might make sense to check the URL character count and request that users select fewer runs if it exceeds a certain limit. 
I think 50 would be a reasonable maximum number of runs. So, 50 runs * (29 characters per run name + n separator characters) = max URL size before Dashi requests you choose fewer runs, which should be much less than the Firefox & Chrome URL size limit (~65000 characters) (though we should make sure there isn't a Jira URL size limit).

**Review by slazicoicr — COMMENTED** [review]
* To save vertical space, would it make sense to have the `Update` and `File Ticket` buttons in the same row?
* I would include the view from which the ticket was filed in the JIRA Summary field: `Dashi RNA-Seq Single Lane: `

**Comment by asofmosq** [comment]
> To save vertical space, would it make sense to have the Update and File Ticket buttons in the same row?

It hasn't been included in this upcoming release but for the single-lane pages, when only a single run is selected an "Approve this run in MISO" button already appears next to the Update button. 

> I would include the view from which the ticket was filed in the JIRA Summary field: Dashi RNA-Seq Single Lane: 

Will update with this amendment.

**Review by callunity — APPROVED** [review]

File: application/dash_application/utility/sidebar_utils.py [hunk 1]
```diff
@@ -316,6 +316,39 @@ def parse_query(query) -> List[str]:
...
+        description = {"description": run_string}
+        return base_url + "&" + urllib.parse.urlencode(description)
+    else:
+        return base_url
+
+
+def jira_display_button(runs: List[str], page_title: str):
+    """ Don't display "File a ticket about these runs" button if more than 100 runs are selected.
+     This prevents an HTTP 400 "Request Header Too Large" error """
+    if len(runs) <= 100:
```
Comment (avarsava) [inline application/dash_application/utility/sidebar_utils.py]: When 0 runs are selected and 'Update' is pressed, 'File a Ticket About These Runs' still shows up

**Review by avarsava — APPROVED** [review]

**Pull request merged**